projeto-volt/src/main/java/sptech/school/projetovolt/api/tag/controller/TagProdutoController.java

```diff
@@ -0,0 +1,53 @@
+package sptech.school.projetovolt.api.tag.controller;
+
+import jakarta.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import sptech.school.projetovolt.entity.tag.TagProduto;
+import sptech.school.projetovolt.entity.tag.dto.TagProdutoConsultaDto;
+import sptech.school.projetovolt.entity.tag.dto.TagProdutoCriacaoDto;
+import sptech.school.projetovolt.entity.tag.dto.TagProdutoMapper;
+import sptech.school.projetovolt.entity.tag.repository.TagProdutoRepository;
+
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/new/tag")
+public class TagProdutoController {
+    @Autowired
+    private TagProdutoRepository tagProdutoRepository;
+
+    @PostMapping
+    public ResponseEntity<TagProdutoConsultaDto> criarTag(@RequestBody TagProdutoCriacaoDto tag){
+        Optional<TagProduto> tagOpt = tagProdutoRepository.findById(TagProdutoMapper.toEntity(tag).getId());
+        if(!tagOpt.get().getTag().equalsIgnoreCase(tag.getTag())){
+            TagProduto tagSalva = tagProdutoRepository.save(TagProdutoMapper.toEntity(tag));
+            return ResponseEntity.status(201).body(TagProdutoMapper.toDto(tagSalva));
+        }
+        return ResponseEntity.status(409).build();
+    }
+
+    @PatchMapping("/{id}")
+    public ResponseEntity<TagProdutoConsultaDto> alterarTag(@PathVariable int id, @RequestParam @Valid String tag){
+        Optional<TagProduto> tagParaAlterar = tagProdutoRepository.findById(id);
+        if(tagParaAlterar.isPresent()){
+            TagProduto alterado = tagParaAlterar.get();
+            alterado.setTag(tag);
+            tagProdutoRepository.save(alterado);
+
+            return ResponseEntity.status(200).body(TagProdutoMapper.toDto(alterado));
+        }
+        return ResponseEntity.status(404).build();
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deletarTag(@PathVariable int id){
+        Optional<TagProduto> encontrado = tagProdutoRepository.findById(id);
+        if(encontrado.isPresent()){
+            tagProdutoRepository.delete(encontrado.get());
+            return ResponseEntity.status(204).build();
+        }
+        return ResponseEntity.status(404).build();
+    }
```

Comment:
## Erro 500 ao deletar com id válido

### Envio
URL: `http://localhost:8080/new/tag/1`;

### JSON de Resposta
```json
{
  "timestamp": "2024-04-19T23:10:53.288+00:00",
  "status": 500,
  "error": "Internal Server Error",
  "message": "could not execute statement [Cannot delete or update a parent row: a foreign key constraint fails (`bd_volt`.`tb_classificacao_produto`, CONSTRAINT `tb_classificacao_produto_ibfk_1` FOREIGN KEY (`fk_tag_produto`) REFERENCES `tb_tag_produto` (`id`))] [delete from tb_tag_produto where id=?]; SQL [delete from tb_tag_produto where id=?]; constraint [null]",
  "path": "/new/tag/1"
}
```

Reply:
<h3><b> Erro ainda continua </b></h3>
<p>
O erro ainda persiste em continuar ao utilizar o endpoint
</P>
<h4> <b> JSON de resposta </b></h4>
{
	"timestamp": "2024-04-21T22:43:57.148+00:00",
	"status": 500,
	"error": "Internal Server Error",
	"message": "could not execute statement [Cannot delete or update a parent row: a foreign key constraint fails (`bd_volt`.`tb_classificacao_produto`, CONSTRAINT `tb_classificacao_produto_ibfk_1` FOREIGN KEY (`fk_tag_produto`) REFERENCES `tb_tag_produto` (`id`))] [delete from tb_tag_produto where id=?]; SQL [delete from tb_tag_produto where id=?]; constraint [null]",
	"path": "/new/tag/1"
}

Reply:
<h3><b>Errro corrigido </b></h3>
<p> Erro foi corrigido por correção no banco de dados, permitindo eliminar uma tag que possuía relação com outra tabela
</p>